src/v2i-hub/ODEForwardPlugin/src/ODEForwardPlugin.cpp

```diff
@@ -148,13 +148,14 @@
 
         //  check for schedule
         if(_freqCounter++%_scheduleFrequency == 0)
-                QueueKafkaMessage(kafka_producer, _BSMkafkaTopic, teststring);	
-	
+                QueueKafkaMessage(kafka_producer, _BSMkafkaTopic, teststring);
+		
+		delete teststring;
```

Comment:
@jwillmartin This delete should be outside of this control block as the memory should be freed regardless of whether or not it is sent to Kafka.

Reply:
@rob-baily My bad, forgot to include the brackets on the if statement. Just pushed changes.

Reply:
Ah I see that yours was actually fine but I do think that having the brackets helps make it clear..